src/Taboo/Turn.py

```diff
@@ -0,0 +1,82 @@
+"""Represents a single Turn in Taboo"""
+
+from enum import Enum
+
+from fwk.MsgSrc import (
+        ConnectionsGroup,
+        Jmai,
+        MsgSrc,
+)
+
+class TurnState(Enum):
+    IN_PLAY = 1
+    COMPLETED = 2
+    DISCARDED = 3
+    TIMED_OUT = 4
+
+class Turn:
+    def __init__(self, turnIdx, wordIdx,
+                 secret, disallowed,
+                 player, otherTeams, allConns,
+                 state=TurnState.IN_PLAY,
+                 score=None):
+        self.turnIdx = turnIdx
+        self.wordIdx = wordIdx
+        self.secret = secret
+        self.disallowed = disallowed
+        self.player = player
+        self.otherTeams = otherTeams
+        self.allConns = allConns
+        self.state = state
+        self.score = score or [] # list of teamNumbers that should be awarded points
+
+        # Messages are broadcast to everyone connected to the room
+        self.publicMsgSrc = MsgSrc(self.allConns)
+
+        # If we are in play, we have to send private messages revealing
+        # The word in play to some people
+        self.privateMsgSrc = None
+        if state == TurnState.IN_PLAY:
+            self.privateConnsGrp = ConnectionsGroup()
```

Comment:
do we need to make it a class-attribute?  Can it not be local?

Reply:
this can't be local (we discussed offline)